Marlin/Marlin_main.cpp

```diff
@@ -2442,10 +2440,7 @@ static void homeaxis(AxisEnum axis) {
       sync_plan_position();
 
       // Move to the adjusted endstop height
-      feedrate = homing_feedrate[axis];
-      destination[Z_AXIS] = adj;
-      line_to_destination();
-      stepper.synchronize();
+      do_blocking_move_to_z(adj, homing_feedrate[axis]);
```

Comment:
It does
```
  #if ENABLED(DELTA)
    feedrate = (feed_rate != 0.0) ? feed_rate : XY_PROBE_FEEDRATE;
    if (x == current_position[X_AXIS] && y == current_position[Y_AXIS])
      prepare_move_to_destination_raw(); // this will also set_current_to_destination !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
    else
      prepare_move_to_destination();     // this will also set_current_to_destination !!!!!!!!!!!!!!!!!!!!!!!!!
  #else
    if (current_position[Z_AXIS] < z) {
      feedrate = (feed_rate != 0.0) ? feed_rate : homing_feedrate[Z_AXIS];
      current_position[Z_AXIS] = z; // !!!!!!!!!!!!!!!!!!!
      line_to_current_position();
    }

    feedrate = (feed_rate != 0.0) ? feed_rate : XY_PROBE_FEEDRATE;
    current_position[X_AXIS] = x;  // !!!!!!!!!!!!
    current_position[Y_AXIS] = y;  // !!!!!!!!!!!!
    line_to_current_position();

    // If Z needs to lower, do it after moving XY
    if (current_position[Z_AXIS] > z) {
      feedrate = (feed_rate != 0.0) ? feed_rate : homing_feedrate[Z_AXIS];
      current_position[Z_AXIS] = z;
      line_to_current_position(); // !!!!!!!!!!!!!!!!!!!!!
    }

  #endif
```